core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -2435,6 +2445,21 @@ private[zio] trait ZIOFunctions extends Serializable {
    */
   final val none: UIO[Option[Nothing]] = succeed(None)
 
+  /**
+   * Feeds elements of type A to a function f that returns an effect.
+   *
+   * Collects all successes and failures in a tupled fashion.
+   */
+  final def partitionM[R, E, A, B](in: Iterable[A])(f: A => ZIO[R, E, B]): ZIO[R, Nothing, (List[E], List[B])] =
```

Comment:
I think we could make this more readable by destructuring the tuple.

```scala
final def partitionM[R, E, A, B](in: Iterable[A])(f: A => ZIO[R, E, B]): ZIO[R, Nothing, (List[E], List[B])] =
  ZIO.foldRight(in)(List.empty[E] -> List.empty[B]) {
    case (a, (es, bs)) =>
      f(a).fold(
        e => (e :: es, bs),
        b => (es, b :: bs)
      )
  }
```

Reply:
Oh damn! That is brilliant. 💯 💯